autoprotocol/builders.py

```diff
@@ -1557,38 +1581,101 @@ def device_mode_params(
         Raises
         ------
         ValueError
-            If model is not None or "high_volume"
+            If device is not of the following: [x_tempest_chip, x_mantis]
+        """
+        params: dict = {
+            "model": model,
+            "material": chip_material,
+            "nozzle": nozzle,
+            "diaphragm": diaphragm,
+            "nozzle_size": nozzle_size,
+            "tubing": tubing,
+            "z_drop": z_drop,
+            "viscosity": viscosity,
+        }
+        if device not in ["x_mantis", "x_tempest_chip"]:
+            raise ValueError(
+                f"Device is {device}. It must be: [x_tempest_chip, x_mantis]"
+            )
+
+        default_params = LiquidHandleBuilders.validate_device_params(device, params)
+
+        device_dict: dict = {}
+        if any(params.values()):
```

Comment:
so, if there are any values, we iterate through and assign _something_ to each key

but what if there are no values? we just pass an empty dict?
should we do similar behavior for any key that doesn't have a value?

Reply:
Yeah, depending on the device, there's a default device dict! Right now it's defaulted to `x_tempest_chip`, but I can set it to a required arg and make it backwards-compatible!

Reply:
This is more so aligned with the original code; if we want to redo that logic, then I feel like that might be a much larger lift than expected 😭 I am more than happy to work on it, just also want to make sure it's within our scope